pkg/networkmanager/firewall.jsx

```diff
@@ -777,7 +800,39 @@ export class Firewall extends React.Component {
     }
 
     onRemoveZone(zone) {
-        firewall.deactiveateZone(zone);
+        let body;
+        if (firewall.zones[zone].services.indexOf("cockpit") !== -1)
+            body = <Alert type="warning">{_("This zone contains the cockpit service. Removing the zone risks making the web console unreachable.")}</Alert>;
+        this.setState({
+            deleteConfirmationModal: <DeleteConfirmationModal title={ cockpit.format(_("Remove zone $0"), zone) }
+            body={body}
+            onCancel={ () =>
+                this.setState({ deleteConfirmationModal: undefined })
+            }
+        onDelete={ () => {
+            firewall.deactiveateZone(zone);
+            this.setState({ deleteConfirmationModal: undefined });
+        }} />
+        });
+    }
+
+    onRemoveService(zone, service) {
+        if (service === 'cockpit') {
+            const body = <Alert type="warning">{_("Removing the cockpit service might result in the web console becoming unreachable. Make sure that the zone you're removing the cockpit service from doesn't apply to your current connection with the web console.")}</Alert>;
```

Comment:
Let's please tidy up the grammar here -- no dangling prepositions and no contractions. Also, the user chose a specific zone and the dialog mentions it, so IMHO this is shorter, less confusing, and grammatically better:

".. unreachable. Make sure that this zone does not apply to your current web console connection."